src/main/java/org/broadinstitute/hellbender/engine/spark/SparkCommandLineProgram.java

```diff
@@ -85,4 +93,36 @@ protected void afterPipeline(final JavaSparkContext ctx) {
     protected String getProgramName(){
         return programName == null ? getClass().getSimpleName() : programName;
     }
+
+    /**
+     * Returns the number of cores being used by each worker.
+     * In local mode it will return the total number of cores allocated to the process.
+     */
+    protected int getExecutorCores() {
```

Comment:
I think we could replace this method with a single line:
`return ctx.defaultParallelism();`
My testing seems to suggest this gives the right answer locally and on a cluster.  One big gotcha, however: there's a race condition where if you call it too quickly on a brand new JavaSparkContext when on a real cluster, it gives a bogus answer.  I have no idea what to do about that except to document it.